app/models/like.rb

```diff
@@ -0,0 +1,6 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :user
+
+  validates :user_id, uniqueness: {scope: :post_id, message: "post already liked by the user"}
```

Comment:
please read about the pitfalls of using uniqueness check on the application side, esp with concurrent requests.